frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/RebaseCurrentBranchOntoParentAction.java

```diff
@@ -21,11 +22,20 @@ protected void onUpdate(AnActionEvent anActionEvent) {
       return;
     }
 
-    var isNonRootBranch = getCurrentBranchNameIfManaged(anActionEvent)
-        .flatMap(bn -> getManagedBranchByName(anActionEvent, bn))
-        .map(b -> b.isNonRoot())
-        .getOrElse(false);
+    var currentBranch = getCurrentBranchNameIfManaged(anActionEvent)
+        .flatMap(bn -> getManagedBranchByName(anActionEvent, bn)).getOrNull();
 
-    presentation.setVisible(isNonRootBranch);
+    if (currentBranch != null) {
+      var isNonRootBranch = currentBranch.isNonRoot();
+
+      if (isNonRootBranch) {
+        var isNotMerged = currentBranch.asNonRoot().getSyncToParentStatus() != SyncToParentStatus.MergedToParent;
```

Comment:
Possibly changing to:
```suggestion
      if (currentBranch.isNonRoot()) {
        var isNotMerged = currentBranch.asNonRoot().getSyncToParentStatus() != SyncToParentStatus.MergedToParent;
```
will fix subtyping checker